packages/eslint-plugin/src/rules/promise-function-async.ts

```diff
@@ -102,6 +102,10 @@ export default util.createRule<Options, MessageIds>({
         return;
       }
 
+      if (node.parent && node.parent.type === 'Property') {
```

Comment:
Need to tighten this check, or else it'll ignore all property functions as well:
```ts
const x = {
  get foo() {}, // parent === Property
  set foo() {}, // parent === Property
  bar: () => {}, // parent === Property
  baz() {}, // parent === Property
}
```

```suggestion
      if (
        node.parent &&
        node.parent.type === 'Property' &&
        (node.parent.kind === 'get' || node.parent.kind === 'set')
      ) {
```